mdoc/Mono.Documentation/Updater/Formatters/FSharpFormatter.cs

```diff
@@ -0,0 +1,1084 @@
+        {
+            AppendAbstractMethod(buf, method, visibility);
+            buf.Append(GetLineEnding());
+            AppendOverrideMethod(buf, method, visibility);
+        }
+
+        private void AppendStaticMethod(StringBuilder buf, MethodDefinition method, string visibility)
+        {
+            buf.Append("static member ");
+            buf.Append(visibility);
+            AppendMethodDeclarationEnding(buf, method);
+        }
+
+        private void AppendAbstractMethod(StringBuilder buf, MethodDefinition method, string visibility)
+        {
+            buf.Append("abstract member ");
+            buf.Append(visibility);
+            AppendMethodDeclarationEnding(buf, method);
+        }
+
+        private void AppendOverrideMethod(StringBuilder buf, MethodDefinition method, string visibility)
+        {
+            buf.Append("override ");
+            buf.Append(visibility);
+            buf.Append("this.");
+            AppendMethodDeclarationEnding(buf, method);
+        }
+
+        private void AppendCommonMethod(StringBuilder buf, MethodDefinition method, string visibility)
+        {
+            buf.Append("member ");
+            buf.Append(visibility);
+            buf.Append("this.");
+            AppendMethodDeclarationEnding(buf, method);
+        }
+
+        private void AppendMethodDeclarationEnding(StringBuilder buf, MethodDefinition method)
+        {
+            AppendMethodName(buf, method);
+            if (!IsOperator(method))
+                AppendGenericMethod(buf, method);
+            buf.Append(" : ");
+            AppendFunctionSignature(buf, method);
+        }
+
+        protected override StringBuilder AppendMethodName(StringBuilder buf, MethodDefinition method)
+        {
+            if (IsOperator(method))
+            {
+                // this is an operator
+                if (TryAppendOperatorName(buf, method))
+                    return buf;
+
+                return base.AppendMethodName(buf, method);
+            }
+
+            var compilationSourceNameAttribute = method.CustomAttributes.FirstOrDefault
+                (i => i.AttributeType.FullName == "Microsoft.FSharp.Core.CompilationSourceNameAttribute");
+            if (compilationSourceNameAttribute != null)
+                return buf.Append(compilationSourceNameAttribute.ConstructorArguments.First().Value);
+
+            return buf.Append(method.Name);
+        }
+
+        protected override StringBuilder AppendGenericMethodConstraints(StringBuilder buf, MethodDefinition method)
+        {
+            return buf;
+        }
+
+        protected override StringBuilder AppendRefTypeName(StringBuilder buf, TypeReference type, DynamicParserContext context)
+        {
+            return buf;
+        }
+
+        protected override string GetFinalizerName(MethodDefinition method)
+        {
+            throw new NotSupportedException("No finalizers in F#");
```

Comment:
Should this throw an exception? if we try to process an assembly that contains a finalizer, is the `mdoc` process going to error out? I would prefer the formatter to return false from `IsSupported` on a finalizer member, rather than throwing an exception here ...

In fact, I'd like to see a finalizer as part of the integration tests to test this very scenario for the F# signatures.

Reply:
Removed this method. It was never called for F#